packages/components/src/popover/index.js

```diff
@@ -490,18 +491,25 @@ const Popover = (
 	] );
 
 	// Event handlers
-	const maybeClose = ( event ) => {
-		// Close on escape
-		if ( event.keyCode === ESCAPE && onClose ) {
-			event.stopPropagation();
-			onClose();
-		}
+	useEffect( () => {
```

Comment:
For event handlers, we should use `useRefEffect` to guarantee event handlers get reattached when the ref changes.

Reply:
Makes sense!